IntegrationTests/tests_04_performance/test_01_resources/shared.swift

```diff
@@ -136,3 +136,133 @@ func withAutoReleasePool<T>(_ execute: () throws -> T) rethrows -> T {
     }
     #endif
 }
+
+// MARK: UDP Shared
+
+/// Repeatedly send, echo and receive a number of UDP requests.
+/// Based on the UDPEchoClient/Server example.
+enum UdpShared {
+    private final class EchoHandler: ChannelInboundHandler {
+        public typealias InboundIn = ByteBuffer
+        public typealias OutboundOut = ByteBuffer
+
+        public func channelRead(context: ChannelHandlerContext, data: NIOAny) {
+            // As we are not really interested getting notified on success or failure we just pass nil as promise to
+            // reduce allocations.
+            context.write(data, promise: nil)
+        }
+
+        public func channelReadComplete(context: ChannelHandlerContext) {
+            // As we are not really interested getting notified on success or failure we just pass nil as promise to
+            // reduce allocations.
+            context.flush()
+        }
+
+        public func errorCaught(context: ChannelHandlerContext, error: Error) {
+            print("error: ", error)
+
+            // As we are not really interested getting notified on success or failure we just pass nil as promise to
+            // reduce allocations.
+            context.close(promise: nil)
+        }
+    }
+
+    private final class EchoHandlerClient: ChannelInboundHandler {
+        public typealias InboundIn = AddressedEnvelope<ByteBuffer>
+        public typealias OutboundOut = AddressedEnvelope<ByteBuffer>
+        private var numBytesOutstanding = 0
+        private var repetitionsRemaining : Int
+        
+        private let remoteAddressInitializer: () throws -> SocketAddress
+        
+        init(remoteAddressInitializer: @escaping () throws -> SocketAddress,
+             numberOfRepetitions: Int) {
+            self.remoteAddressInitializer = remoteAddressInitializer
```

Comment:
Why do we delay this?

Reply:
I just copied the logic from the existing example - ie, I don't know.
I could try without and see what happens.

Reply:
Made direct and it worked just fine.